aiohttp/client.py

```diff
@@ -1020,8 +1020,13 @@ def __init__(self,
         self._session = session
 
     async def __aenter__(self) -> ClientResponse:
-        self._resp = await self._coro
-        return self._resp
+        try:
+            self._resp = await self._coro
+        except BaseException:
```

Comment:
I think that this is too broad. Best practice is to catch a specific exception class.

Reply:
It's going to be reraised shortly and used just to cleanup resources. It's legal usage for such a case. Actually, we have similar catches in other places, but reraise happens always shortly. Certainly, if there possible to limit the scope by some OSError - that would be great, but this may keep the bug for future changes when other exception types will be raised as well.

Reply:
Don't we also need to catch KeyboardInterrupt (or maybe even GeneratorExit) here?

Reply:
If yes, that would leave us with 
```
try:
    ...
except (*bunch_of_base_exceptions, *bunch_of_os_exceptions)
    ...
```
It might be nice if we could narrow down all exceptions that we expect in `session._request` to just one class (so `*bunch_of_os_exceptions` is actually just one exception).
But there is another problem: i believe we can't  make everything  foolproof and there will always be the case when user supplied some invalid arguments, which resulted in unforeseen exception (TypeError, ValueError, who knows) inside aiohttp. I think in such case it is more reasonable to cleanup resources "no matter what" and re-raise.

Reply:
> Don't we also need to catch KeyboardInterrupt (or maybe even GeneratorExit) here?
It doesn't matters since you do `raise` a little bit latre.

Reply:
I mean in case we do it the way @webknjaz suggested (catch specific exception class). One class won't be enough, we will need to catch at least KeyboardInterrupt and may be several other classes.

Reply:
Going to resolve this. If anyone has further arguments against BaseException please bring them on.

Reply:
Agreed.